src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -106,12 +106,51 @@ class TestConcurrencyManager : public TestLoadManagerBase,
   ///
   void TestSequences()
   {
-    auto sleep_time = std::chrono::milliseconds(500);
+    int delay_ms = 10;
+    stats_->response_delay = std::chrono::milliseconds(delay_ms);
+
+    auto stats = cb::InferStat();
+    double concurrency1 = params_.max_concurrency / 2;
+    double concurrency2 = params_.max_concurrency;
+    int sleep_ms = 500;
+    double num_seconds = sleep_ms / 1000;
+
+    auto sleep_time = std::chrono::milliseconds(sleep_ms);
+    size_t expected_count1 = sleep_ms * concurrency1 / delay_ms;
+    size_t expected_count2 =
+        sleep_ms * concurrency2 / delay_ms + expected_count1;
+
+    // Run and check request rate 1
+    //
+    ChangeConcurrencyLevel(concurrency1);
+    std::this_thread::sleep_for(sleep_time);
 
-    ChangeConcurrencyLevel(params_.max_concurrency);
+    stats = cb::InferStat();
+    GetAccumulatedClientStat(&stats);
+    CHECK(
+        stats.completed_request_count ==
```

Comment:
Does this not get reset between changes to concurrency?

Reply:
Yes and no. The client just keeps a running tally. The load manager takes checkpoints of it at various times to determine start and end values for a given window.  This is all related to the grpc time bug that will be fixed in my next story.